Composer/packages/lib/shared/__tests__/copyUtils/copyAdaptiveAction.test.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { copyAdaptiveAction } from '../../src/copyUtils';
+import { CopyConstructorMap } from '../../src/copyUtils/copyAdaptiveAction';
+import { externalApiStub as externalApi } from '../jestMocks/externalApiStub';
+import { SDKTypes } from '../../src';
+
+describe('copyAdaptiveAction', () => {
+  it('should return {} when input is invalid', async () => {
+    expect(await copyAdaptiveAction('hello', externalApi)).toEqual('hello');
+
+    expect(await copyAdaptiveAction(null as any, externalApi)).toEqual({});
+    expect(await copyAdaptiveAction({} as any, externalApi)).toEqual({});
+    expect(await copyAdaptiveAction({ name: 'hi' } as any, externalApi)).toEqual({});
+  });
+
+  describe('should call certain handler in CopyConstructorMap', () => {
+    let originCopyConstructorMap = {};
+    beforeAll(() => {
+      originCopyConstructorMap = { ...CopyConstructorMap };
+    });
+
+    beforeEach(() => {
+      Object.keys(CopyConstructorMap).forEach(key => {
+        CopyConstructorMap[key] = jest.fn();
+      });
+    });
+
+    afterAll(() => {
+      Object.keys(CopyConstructorMap).forEach(key => {
+        CopyConstructorMap[key] = originCopyConstructorMap[key];
+      });
+    });
+
+    it('when handler registered', async () => {
+      const registeredUniqueTypes = [
+        SDKTypes.SendActivity,
+        SDKTypes.IfCondition,
+        SDKTypes.SwitchCondition,
+        SDKTypes.EditActions,
+        SDKTypes.ChoiceInput,
+        SDKTypes.Foreach,
+      ];
+
+      for (const $type of registeredUniqueTypes) {
+        await copyAdaptiveAction({ $type }, externalApi);
+        expect(CopyConstructorMap[$type]).toHaveBeenCalledTimes(1);
+      }
+
+      for (const $type of registeredUniqueTypes) {
+        await copyAdaptiveAction({ $type }, externalApi);
+        expect(CopyConstructorMap[$type]).toHaveBeenCalledTimes(2);
+      }
```

Comment:
I don't get this. What is it testing?

Reply:
just to make sure copiers are called every time. I will remove it because of it's not necessary